rioxarray/xarray_plugin.py

```diff
@@ -0,0 +1,59 @@
+import os.path
+
+import xarray as xr
+
+from . import _io
+
+CAN_OPEN_EXTS = {
+    "asc",
+    "geotif",
+    "geotiff",
+    "img",
+    "j2k",
+    "jp2",
+    "jpg",
+    "jpeg",
+    "png",
+    "tif",
+    "tiff",
+    "vrt",
+}
+
+
+class RasterioBackend(xr.backends.common.BackendEntrypoint):
+    """
+    .. versionadded:: 0.4
+    """
+
+    def open_dataset(
+        self,
+        filename_or_obj,
+        drop_variables=None,
+        mask_and_scale=True,
+        parse_coordinates=None,
+        lock=None,
+        masked=False,
+        variable=None,
+        group=None,
+        default_name="band_data",
```

Comment:
I noticed `**open_kwargs` is missing here. This is nice to have to be able to pass in custom arguments to `rasterio.open`.

There are two paths forward for this:
1. Only add `**open_kwargs`.
2. Remove all arguments except `filename_or_obj`, `mask_and_scale=True` and `defaut_name` arguments and use `**open_kwargs` as a catch all to pass in the rest of the arguments to `open_rasterio`.

I would be fine with either way. I am leaning towards 2 at the moment because it makes it so you don't have to update the `RasterioBackend` every time you update `open_rasterio`.

Reply:
The backend API insists in having all keyword arguments explicit and no `**kwargs` in order to raise user friendly error messages when an argument is not supported, so option 2 is off the table and option 1 is a bit ugly.

I'll implement option 1. ugly.